api/v2/api.go

```diff
@@ -389,6 +341,143 @@ func (api *API) postAlertsHandler(params alert_ops.PostAlertsParams) middleware.
 	return alert_ops.NewPostAlertsOK()
 }
 
+func (api *API) getAlertGroupsHandler(params alertgroup_ops.GetAlertGroupsParams) middleware.Responder {
+	var (
+		err            error
+		receiverFilter *regexp.Regexp
+		matchers       = []*labels.Matcher{}
+	)
+
+	for _, matcherString := range params.Filter {
+		matcher, err := parse.Matcher(matcherString)
+		if err != nil {
+			level.Error(api.logger).Log("msg", "failed to parse matchers", "err", err)
+			return alertgroup_ops.NewGetAlertGroupsBadRequest().WithPayload(err.Error())
+		}
+
+		matchers = append(matchers, matcher)
+	}
+
+	if params.Receiver != nil {
+		receiverFilter, err = regexp.Compile("^(?:" + *params.Receiver + ")$")
+		if err != nil {
+			return alertgroup_ops.
```

Comment:
In the above `for` loop we log the error, here we don't. Can we be consistent with either way?